infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -436,42 +436,6 @@ private void initBilling() {
         }
     }
 
-    @Override
-    protected void unregisterNodeAndRemoveInstanceIfNeeded(final String instanceId, final String nodeName,
-            final String infrastructureId, final boolean terminateInstanceIfEmpty) {
-        setPersistedInfraVariable(() -> {
-            // first read from the runtime variables map
-            nodesPerInstance = (Map<String, Set<String>>) persistedInfraVariables.get(NODES_PER_INSTANCES_KEY);
-            // make modifications to the nodesPerInstance map
-            if (nodesPerInstance.get(instanceId) != null) {
-                nodesPerInstance.get(instanceId).remove(nodeName);
-                LOGGER.info("Removed node: " + nodeName);
-                if (nodesPerInstance.get(instanceId).isEmpty()) {
-                    if (terminateInstanceIfEmpty) {
-                        connectorIaasController.terminateInstance(infrastructureId, instanceId);
```

Comment:
Shouldn't the instance be terminated ?

Reply:
This method already exists in the mother class.